### ISSUE TITLE
Game Guru classic | the 3rd person mode should be done

### ISSUE THREAD

**Issue by Peri32** [Dark-Basic-Software-Limited/GameGuruRepo#3459]
Hi there, 

Recently and in recent 3 years I have been thinking a lot about how to bring the third person mode of the classic Game Guru to match the abilities of the first person, basically what is missing is the ability to swap weapons, muzzle flash, duck, reload animation

I am not a professional programmer and I know the main team focuses on the first person, so I thought to hire a programmer for this project (yes I am ready to invest money)  and back to the main team and suggest them to include it in the next updates if they think it's good enough.

one of my ideas is also to make the character not always in point the gun position but only when we use the right mouse.

I  try to start and understand where to start before I talk with programmers, If you are interested in guiding me I would be happy.

**Comment by davetgc** [comment]
3rd person is a big job, more complicated then many realise if it's to be done correctly, and isn't planned for v1. A true 3rd person mode needs to tie in with a lot of areas, made more complicated by the additions of inventories, etc.

While it's on the list, we'd rather that it was done properly and consider getting 1sr person fully integrated first a priority.

**Comment by Peri32** [comment]
Hi, thanks for the answer, I understand you mean to MAX yes I fully understand that but about the classic, I think the first person is in good condition so its possible to complete the 3rd there are the basics, and only a few features are required to create a classic fun 3rd shooter with it, I keep talking about it cause I see the potential. I know the team is focused on MAX so I can help with hiring a programmer and try to add these features mostly I talk about the ability to swap weapons, muzzle flash, duck, reload animation. maybe a new kind of soldier character is required to do it.

**Comment by Peri32** [comment]
Me Smallg and one more person try an alternative method to complete the Tpp of the classic, we use an alternative character for the animations, we need help to complete the ducking action we think there is a capsule around the Tpp character and when we duck it's not really duck we looking for a way to control the capsule size at ducking, can someone give us help with it where can we find it in the code? please watch the video

 https://www.youtube.com/watch?v=Sov2ES0UbyI&ab_channel=EldarPeri

**Comment by LeeBamberTGC** [comment]
@Peri32 Yes you are seeing the size of the physics capsule created for characters which we never expected to duck. The player capsule can duck just fine, and in the repo code you can see how we resize the capsule on the fly when the crouch mode is needed. Something similar will be needed to change the character entity physics shape in real-time and in the same way so the collision is maintained with the floor and other objects, but will allow the character (TPP) to duck under things.  Raid the code and look for 'capsules' in and around the physics areas, you should find the player, and can do a little cut and paste to give the same power to arbitrary character entities (including the TPP), perhaps under a new collision mode.

**Comment by Peri32** [comment]
I see, thanks for the answer we will try to focus on another features.

**Comment by synchromesh62** [comment]
Thats quite impressive what you have managed so far

**Comment by Peri32** [comment]
> Thats quite impressive what you have managed so far 

Thanks :D I strongly believe the Tpp mode can supply a different experience to Game Guru and it have high potential to bring in more creators, I will do my best with my alternative solution

**Comment by Peri32** [comment]
Hello guys, we have some nice progress with few features, I hope to finish it and share it with the community

I need help to locate the tpp melee function of the unarmed characters, my goal is to activate the melee with the key Q even if the character have a gun.

I hope its not hard coded thing.

**Comment by Peri32** [comment]
@LeeBamberTGC 

Hello guys, we have some nice progress with few features, I hope to finish it and share it with the community

I need help to locate the tpp melee function of the unarmed characters, my goal is to activate the melee with the key Q even if the character have a gun.

I hope its not hard coded thing.

**Comment by Blammo0507** [comment]
Hello everyone!
I’ve been waiting for a long time for a third-person player update or expansion for Game Guru Classic. The close-combat animations and swimming – I think that would really give Classic a boost... Could there perhaps be a final major update featuring these features? I think a lot of people would be delighted! Thank you, and apologies for my poor English...

**Comment by Kasseyus** [comment]
Unfortunately, there is no active development of features on GameGuru Classic beyond bug fixes so there will not be any official patches to add this functionality.
